spec/haskell/src/SEL4/Kernel/FaultHandler.lhs

```diff
@@ -33,11 +33,22 @@ This module contains functions that determine how recoverable faults encountered
 
 > isValidTimeoutHandler :: PPtr TCB -> Kernel Bool
 > isValidTimeoutHandler tptr = do
->     tcb <- getObject tptr
->     case cteCap (tcbTimeoutHandler tcb) of
+>     stateAssert (tcb_at'_asrt tptr)
+>         "Assert that there is a TCB at tptr"
+>     timeoutHandlerSlot <- getThreadTimeoutHandlerSlot tptr
+>     timeoutHandlerCap <- getSlotCap timeoutHandlerSlot
+>     case timeoutHandlerCap of
 >         EndpointCap {} -> return True
 >         _ -> return False
 
+These functions will return a physical pointer to a thread's fault or timeout handler slot.
```

Comment:
why "will return"? you use that phrasing for assert statements whose definitions will be defined later... but the functions do return a pointer? I'd suggest:

`These functions return a physical pointer to a thread's fault and timeout handler slot respectively.`

Reply:
It's the same text as for the existing getSlot functions. If we are changing it, I would prefer not to call them physical pointers, because that whole PPtr terminology in Haskell has always been terminally confusing once things hit virtual memory. I'd just say "a pointer".